connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectCluster.java

```diff
@@ -90,26 +132,48 @@ public void start() throws IOException {
      * Clean up any temp directories created locally.
      */
     public void stop() {
-        for (Connect worker : this.connectCluster) {
-            try {
-                worker.stop();
-            } catch (Exception e) {
-                log.error("Could not stop connect", e);
-                throw new RuntimeException("Could not stop worker", e);
-            }
-        }
-
+        connectCluster.forEach(this::stopWorker);
         try {
             kafkaCluster.after();
         } catch (Exception e) {
             log.error("Could not stop kafka", e);
             throw new RuntimeException("Could not stop brokers", e);
+        } finally {
+            Exit.resetExitProcedure();
+            Exit.resetHaltProcedure();
+        }
+    }
+
+    /**
+     * Provision and start an additional worker to the Connect cluster.
+     */
+    public void addWorker() {
+        connectCluster.add(new ConnectDistributed().startConnect(workerProps));
+    }
+
+    /**
+     * Decommission a worker from this Connect cluster.
```

Comment:
Nit: it's not clear which worker this will remove, so how about "Decommission one of the workers ..."? 

Also, does it really help to remove an arbitrary worker? This happens to remove the last worker, so would `removeLastWorkerAdded()` be more clear?

Reply:
I wanted to avoid bookkeeping of workers outside this class for the moment. We can always introduce it when we need it. Another candidate for instance will probably be at some point `removeLeaderWorker`. Also, I don't think it's too bad to keep which worker as implementation specific. We don't always have to over-commit in the method contracts. "A worker" can be always the last, a random, or the first. 

nit: I read "one of" and "a" as synonymous, but I'm happy to change.

Reply:
I introduced `WorkerHandler` and looks a bit more functional now with both options to select a worker to stop and stop the "a" worker (still the last one here)